requirements/compile.py

```diff
@@ -8,37 +8,23 @@
 
 if __name__ == "__main__":
     os.chdir(Path(__file__).parent)
+
     os.environ["CUSTOM_COMPILE_COMMAND"] = "requirements/compile.py"
     os.environ["PIP_REQUIRE_VIRTUALENV"] = "0"
-    common_args = [
-        "-m",
-        "piptools",
-        "compile",
-        "--generate-hashes",
-        "--allow-unsafe",
-    ] + sys.argv[1:]
-    subprocess.run(
-        ["python3.7", *common_args, "-o", "py37.txt"],
-        check=True,
-        capture_output=True,
-    )
-    subprocess.run(
-        ["python3.8", *common_args, "-o", "py38.txt"],
-        check=True,
-        capture_output=True,
-    )
-    subprocess.run(
-        ["python3.9", *common_args, "-o", "py39.txt"],
-        check=True,
-        capture_output=True,
-    )
-    subprocess.run(
-        ["python3.10", *common_args, "-o", "py310.txt"],
-        check=True,
-        capture_output=True,
-    )
-    subprocess.run(
-        ["python3.11", *common_args, "-o", "py311.txt"],
-        check=True,
-        capture_output=True,
-    )
+
+    for python_version in ["3.7", "3.8", "3.9", "3.10", "3.11"]:
```

Comment:
this is deliberately left as an unrolled loop - I have a version of this script in each of my projects, with varying requirements. Rather than have loops, and nested loops, I prefer a flat list.

When making a PR restrict changes to relevant places in the code base only.